backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java

```diff
@@ -77,21 +85,41 @@ public class NotificationResource {
     @Inject
     EndpointRepository endpointRepository;
 
+    @Inject
+    KesselAuthorization kesselAuthorization;
+
     @Path(Constants.API_NOTIFICATIONS_V_1_0 + "/notifications")
     public static class V1 extends NotificationResource {
         @GET
         @Path("/eventTypes/{eventTypeId}/behaviorGroups")
         @Produces(APPLICATION_JSON)
         @Operation(summary = "List the behavior groups linked to an event type", description = "Lists the behavior groups that are linked to an event type. Use this endpoint to see which behavior groups will be affected if you delete an event type.")
-        @RolesAllowed(ConsoleIdentityProvider.RBAC_READ_NOTIFICATIONS)
         public List<BehaviorGroup> getLinkedBehaviorGroups(
                 @Context SecurityContext sec,
                 @PathParam("eventTypeId") UUID eventTypeId,
                 @BeanParam @Valid Query query
         ) {
-            String orgId = getOrgId(sec);
+            if (this.backendConfig.isKesselBackendEnabled()) {
+                this.kesselAuthorization.hasPermissionOnResource(sec, WorkspacePermission.NOTIFICATIONS_READ, ResourceType.WORKSPACE, WORKSPACE_ID_PLACEHOLDER);
```

Comment:
Why checking workspace permission then resource permission? isn't it redundant?

Reply:
You are right, I think we can simply check the permissions for the behavior groups, since that will include the workspace permission too.